app/src/main/java/io/neurolab/fragments/StatisticsFragment.java

```diff
@@ -0,0 +1,123 @@
+        thetaValView = rootView.findViewById(R.id.theta_value);
+        lowAlphaValView = rootView.findViewById(R.id.low_alpha_value);
+        highAlphaValView = rootView.findViewById(R.id.high_alpha_value);
+
+        concEmojiView = rootView.findViewById(R.id.concentrate_emoji);
+        sleepEmojiView = rootView.findViewById(R.id.sleep_emoji);
+        calmEmojiView = rootView.findViewById(R.id.calm_emoji);
+        angerEmojiView = rootView.findViewById(R.id.anger_emoji);
+
+        if (StatisticsFragment.parsedData == null)
+            Toast.makeText(getContext(), "No Data", Toast.LENGTH_SHORT).show();
+        else
+            initStatistics();
+        return rootView;
+    }
+
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+    }
+
+    private void initStatistics() {
+        DefaultFFTData defaultFFTData = new DefaultFFTData(3, 4, 2);
+        double[] data = convertToDouble(StatisticsFragment.parsedData);
+        double[][] dfftData = dataForCalculations(data);
+        defaultFFTData.updateFFTData(dfftData);
+        int[] calculatedValues = defaultFFTData.getBinRangeValues();
+
+        deltaValView.setText(String.valueOf(calculatedValues[0]));
+        thetaValView.setText(String.valueOf(calculatedValues[1]));
+        lowAlphaValView.setText(String.valueOf(calculatedValues[2]));
+        highAlphaValView.setText(String.valueOf(calculatedValues[3]));
+
+        concEmojiView.setVisibility(View.VISIBLE);
+        sleepEmojiView.setVisibility(View.VISIBLE);
+        calmEmojiView.setVisibility(View.VISIBLE);
+        angerEmojiView.setVisibility(View.VISIBLE);
+
+        if(calculatedValues[0] < 40000)
+            concEmojiView.setImageResource(R.drawable.ic_sad);
+        if(calculatedValues[1] < 40000)
+            sleepEmojiView.setImageResource(R.drawable.ic_sad);
+        if(calculatedValues[2] < 40000)
+            calmEmojiView.setImageResource(R.drawable.ic_sad);
+        if(calculatedValues[3] > 150000)
+            angerEmojiView.setImageResource(R.drawable.ic_sad);
+
+    }
+
+    private double[][] dataForCalculations(double[] data) {
+        int rows = 2000;
+        int columns = 4;
+        double[][] dfftData = new double[rows][columns];
+        for (int j = 0; j < rows; j++) {
+            for (int i = 0; i < columns; i++) {
+                dfftData[j][i] = data[j * 4 + i];
+            }
+        }
+        return dfftData;
+    }
+
+    private double[] convertToDouble(String[] parsedData) {
+        double[] parsedDoubleData = new double[parsedData.length];
+        int startTrimIndex = 0;
+        int endTrimIndex = 9;
+        for (int i = 0; i < parsedData.length; i++) {
+            if (parsedData[i].length() > 0) {
+                parsedDoubleData[i] = Double.parseDouble(parsedData[i].substring(startTrimIndex, endTrimIndex));
+                if (parsedDoubleData[i] > 5060) {
+                    parsedDoubleData[i] = 5060;
```

Comment:
Why are we limiting the value here?

Reply:
@CloudyPadmal Actually, because some values, in the dataset which when parsed as float/doubles are being converted to into exponential values which are throwing errors while being plotted. So, here i fixed a maximum value after having a look at all the values in the dataset for now. Later, we can create a custom max value finder function according to different datasets. Do let me know, if we can do something else.

Reply:
Ah I see. I'll work on data normalization part #304 . Once that is fixed it'll be fine